prio-graph-scheduler/src/id_generator.rs

```diff
@@ -1,21 +1,19 @@
-use crate::banking_stage::scheduler_messages::TransactionId;
-
 /// Simple reverse-sequential ID generator for `TransactionId`s.
 /// These IDs uniquely identify transactions during the scheduling process.
-pub struct TransactionIdGenerator {
+pub struct IdGenerator {
     next_id: u64,
 }
 
-impl Default for TransactionIdGenerator {
+impl Default for IdGenerator {
     fn default() -> Self {
         Self { next_id: u64::MAX }
     }
 }
 
-impl TransactionIdGenerator {
-    pub fn next(&mut self) -> TransactionId {
+impl IdGenerator {
+    pub fn next<T: From<u64>>(&mut self) -> T {
         let id = self.next_id;
         self.next_id = self.next_id.wrapping_sub(1);
-        TransactionId::new(id)
+        T::from(id)
```

Comment:
There's actually a subtle difference between `BatchIdGenerator` and `TransactionIdGenerator`, and that's in the implementation of the `Default` trait for each. The default value for `next_id` is zero in `BatchIdGenerator` (from the derive macro), but it's `u64::MAX` in  `TransactionIdGenerator`.

Since these two structs don't have any external dependencies, can they be left separate? Each can still be generic over `From<u64>`.